### DISCUSSION TITLE
Why am I use PhysicsComponent to my entity it will have network problems?

### DISCUSSION THREAD

**Discussion by FyWoOcat** [AlmasB/FXGL#1282] [Q&A] [answered]
When I use .with(new PhysicsComponent()) to build an entity, and use this 

var player1 = spawn("player1", 0, 0,0);
 getService(MultiplayerService.class).spawn(connection, player1, "player1");

server.setOnConnected(conn ->
                    {
                        connection = conn;
                        getExecutor().startAsyncFX(this::onser);
                    });

to send entity to client that It can not move on client but can move on server.

I mean I can control the entity on server by use client, but on client I can not see they move, and idea show this

Failed to find updates:  java.net.SocketTimeoutException: connect timed out 

(sorry about my terrible grammar)

**Comment by AlmasB** [comment, marked as answer]
An entity with PhysicsComponent attached is managed by the physics world, not the game world. As far as I can remember, the network-based entities will only modify the game world coordinates. So a possible solution is to remove PhysicsComponent from the client-side since the logic should be done on the server side.

Let me know if this makes sense.

**Reply by FyWoOcat** [reply]
I have use the server to done the PhysicsComponent, that mean I should use getphysicsworld() right?

**Reply by FyWoOcat** [reply]
Wait, I spawn the ground on both sides of the server and client-side. Is thar a problem?
